include/xtensor/xstorage.hpp

```diff
@@ -28,6 +26,18 @@ namespace xt
                                                                                std::input_iterator_tag>::value>::type;
     }
 
+    /********************************************
+     * xtrivial_default_construct implemenation *
+     ********************************************/
+
+    #if !defined(__GNUG__) || defined(_LIBCPP_VERSION) || defined(_GLIBCXX_USE_CXX11_ABI)
+        template <class T>
+        using xtrivially_default_constructible = std::is_trivially_default_constructible<T>;
+    #else
+        template <class T>
+        using xtrivially_default_constructible = std::has_trivial_default_constructor<T>;
+    #endif
```

Comment:
Why moving this form `xutils.hpp` ? This is a general traits, not specific to the storage, and should be moved to `xtl` enventually.